### PR TITLE
DOCS-9987-basic-alerts-email-templates

### PR DESCRIPTION
**Pull request by sarathecoubian**

### CODE REVIEW

File: modules/ROOT/pages/basic-alerts.adoc [hunk 1]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
 
 include::partial$include-alert-severity.adoc[leveloffset=+1]
 
-// add basic alert email templates
+== Basic Alert Email Template
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:43]:
```suggestion
[[alert-variables]]
== Basic Alert Email Template
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 2]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|`${periodTimeUnit}` |Period of time unit
+|`${resource}` a| Name of the item that triggers the alert:
+
+* CloudHub Apps: Name of the app that triggers the alert
+* Hybrid Apps: Name of the resource in the app that triggers the alert
+|`${severity}` | Severity of the alert
+|`${targetState}` |Server state when the alert was triggered
+|`${threshold}` |Count over the period of time that triggers the alert
+|`${value}` | A value associated with an element
+|===
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:125]: In this list we don't have support yet or are removing these variables: 
clusterName
groupName
name
periodTimeUnit

Here is the list of variables that we changed the name for: 
currentValue -> value (value will be applied to all alerts)
memberName -> targetName (name of the server)

File: modules/ROOT/pages/basic-alerts.adoc [hunk 3]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| Alert Type | Alert | Applicable Variables
+2+| All Alerts | `${resource}` `${severity}` `${state}`
+.7+| CloudHub App Alerts |*CPU usage - CloudHub* .2+|`${operator}` `${value}`
+| *Memory usage - CloudHub*
+| *Custom application notification* | `${priority}` `${message}`
+| *Deployment failed* |  `${user}` `${message}`
+| *Deployment success* .2+| `${user}`
+| *Worker not responding*
+| *Exceeds event traffic threshold* | `${period}` `${periodTimeUnit}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:59]:
```suggestion
| *Exceeds event traffic threshold* | `${period}`
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 4]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+
+
+.4+|Hybrid App Alerts | All Hybrid App Alerts| `${name}`
+| *Number of Mule messages* .3+| `${currentValue}` `${threshold}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:63]:
```suggestion
| *Number of Mule messages* .3+| `${value}` `${threshold}`
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 5]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| *Number of errors*
+| *Response time*
+
+.5+|Server Alerts | All Server Alerts | `${name}`
+| *Server CPU usage* .4+|  `${currentValue}` `${comparison}` `${threshold}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:68]:
```suggestion
| *Server CPU usage* .4+|  `${value}` `${comparison}` `${threshold}`
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 6]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| *Server Load average*
+| *Server Memory average*
+| *Server Thread count*
+
+.9+| Server Group Alerts| All Server Group Alerts | `${name}`
+| *Server Group's node CPU usage* .4+|  `${memberName}` `${groupName}` `${currentValue}` `${comparison}` `${threshold}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:74]:
```suggestion
| *Server Group's node CPU usage* .4+|  `${targetName}` `${value}` `${comparison}` `${threshold}`
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 7]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| *Server Group's node Load average*
+| *Server Group's node Memory usage*
+| *Server Group's node Thread count*
+| *Server Group's node connected* .4+| `${memberName}` `${groupName}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:78]:
```suggestion
| *Server Group's node connected* .4+| `${targetName}`
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 8]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| *Server Group's node disconnected*
+| *Server added to Server Group*
+| *Server removed from Server Group*
+
+.9+| Cluster Alerts |All Cluster Alerts| `${name}`
+| *Cluster's node CPU usage* .4+| `${memberName}` `${clusterName}` `${currentValue}` `${comparison}` `${threshold}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:84]:
```suggestion
| *Cluster's node CPU usage* .4+| `${targetName}` `${value}` `${comparison}` `${threshold}`
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 9]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| *Cluster's node Load average*
+| *Cluster's node Memory usage*
+| *Cluster's node Thread count*
+| *Cluster's node disconnected* | `${resource}` `${memberName}` `${clusterName}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:88]:
```suggestion
| *Cluster's node disconnected* | `${resource}` `${targetName}` 
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 10]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+| *Cluster's node connected* .3+| `${memberName}` `${clusterName}`
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:89]:
```suggestion
| *Cluster's node connected* .3+| `${targetName}` 
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 11]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+
+
+[%header%autowidth.spread]
+[[var-defs]]
+.Variable Definitions
+|===
+|Variable |Definition
+|`${alertLink}`|URL of the triggered basic alert
+|`${alertState}`|The state of the alert when it was triggered. Available values are `OK` or `Alerting`.
+|`${clusterName}`|Name of the cluster
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:102]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 12]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|`${comparison}`|Above or Below
+|`${condition}`|The metric for which the alert is triggered
+|`${currentValue}`|Current value of metric that triggers the alert
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:105]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 13]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|`${period}` |Period of time over which to measure
+|`${periodTimeUnit}` |Period of time unit
+|`${resource}` a| Name of the item that triggers the alert:
+
+* CloudHub Apps: Name of the app that triggers the alert
+* Hybrid Apps: Name of the resource in the app that triggers the alert
+|`${severity}` | Severity of the alert
+|`${targetState}` |Server state when the alert was triggered
+|`${threshold}` |Count over the period of time that triggers the alert
+|`${value}` | A value associated with an element
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:124]:
```suggestion
|`${value}` | The value of the metric that triggers the alert
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 14]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|`${alertLink}`|URL of the triggered basic alert
+|`${alertState}`|The state of the alert when it was triggered. Available values are `OK` or `Alerting`.
+|`${clusterName}`|Name of the cluster
+|`${comparison}`|Above or Below
+|`${condition}`|The metric for which the alert is triggered
+|`${currentValue}`|Current value of metric that triggers the alert
+|`${environment}`|The name of the environment in which the resource is deployed
+|`${groupName}`|Name of the server group
+|`${memberName}`|Name of the node in a server group or cluster
+|`${name}` a|Name of the item that triggers the alert:
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:109]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 15]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|===
+|Variable |Definition
+|`${alertLink}`|URL of the triggered basic alert
+|`${alertState}`|The state of the alert when it was triggered. Available values are `OK` or `Alerting`.
+|`${clusterName}`|Name of the cluster
+|`${comparison}`|Above or Below
+|`${condition}`|The metric for which the alert is triggered
+|`${currentValue}`|Current value of metric that triggers the alert
+|`${environment}`|The name of the environment in which the resource is deployed
+|`${groupName}`|Name of the server group
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:107]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 16]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|`${memberName}`|Name of the node in a server group or cluster
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:108]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 17]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+* Servers: Name of the target (server, server group, or cluster) that triggers the alert
+|`${operator}` |An operator, such as `< > =`
+|`${organization}` |The name of the organization that owns the alert
+|`${period}` |Period of time over which to measure
+|`${periodTimeUnit}` |Period of time unit
+|`${resource}` a| Name of the item that triggers the alert:
+
+* CloudHub Apps: Name of the app that triggers the alert
+* Hybrid Apps: Name of the resource in the app that triggers the alert
+|`${severity}` | Severity of the alert
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:121]:
```suggestion
|`${severity}` | Severity of the alert
|`${targetName}`|Name of the node in a server group or cluster
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 18]
```diff
@@ -40,4 +40,86 @@ The following levels of severity are available for basic alerts:
...
+|`${groupName}`|Name of the server group
+|`${memberName}`|Name of the node in a server group or cluster
+|`${name}` a|Name of the item that triggers the alert:
+
+* Hybrid Apps: Name of the app that triggers the alert
+* Servers: Name of the target (server, server group, or cluster) that triggers the alert
+|`${operator}` |An operator, such as `< > =`
+|`${organization}` |The name of the organization that owns the alert
+|`${period}` |Period of time over which to measure
+|`${periodTimeUnit}` |Period of time unit
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:116]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 19]
```diff
@@ -40,4 +40,82 @@ The following levels of severity are available for basic alerts:
...
+| *Server removed from Cluster*
+|===
+
+
+[%header%autowidth.spread]
+[[var-defs]]
+.Variable Definitions
+|===
+|Variable |Definition
+|`${alertLink}`|URL of the triggered basic alert
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:101]:
```suggestion
|`${alertLink}`|URL of the triggered alert
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 20]
```diff
@@ -40,4 +40,82 @@ The following levels of severity are available for basic alerts:
...
+|`${alertState}`|The state of the alert when it was triggered. Available values are `OK` or `Alerting`.
+|`${comparison}`|Above or Below
+|`${condition}`|The metric for which the alert is triggered
+|`${environment}`|The name of the environment in which the resource is deployed
+
+* Hybrid Apps: Name of the app that triggers the alert
+* Servers: Name of the target (server, server group, or cluster) that triggers the alert
+|`${operator}` |An operator, such as `< > =`
+|`${organization}` |The name of the organization that owns the alert
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:110]:
```suggestion
|`${organization}` |Name of the organization that owns the alert
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 21]
```diff
@@ -40,4 +40,82 @@ The following levels of severity are available for basic alerts:
...
+|===
+
+
+[%header%autowidth.spread]
+[[var-defs]]
+.Variable Definitions
+|===
+|Variable |Definition
+|`${alertLink}`|URL of the triggered basic alert
+|`${alertState}`|The state of the alert when it was triggered. Available values are `OK` or `Alerting`.
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:102]:
```suggestion
|`${alertState}`|State of the alert when it was triggered. Available values are `OK` or `Alerting`.
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 22]
```diff
@@ -40,4 +40,82 @@ The following levels of severity are available for basic alerts:
...
+|`${comparison}`|Above or Below
+|`${condition}`|The metric for which the alert is triggered
+|`${environment}`|The name of the environment in which the resource is deployed
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:105]:
```suggestion
|`${condition}`|Metric for which the alert is triggered
|`${environment}`|Name of the environment in which the resource is deployed
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 23]
```diff
@@ -40,4 +40,82 @@ The following levels of severity are available for basic alerts:
...
+
+* Hybrid Apps: Name of the app that triggers the alert
+* Servers: Name of the target (server, server group, or cluster) that triggers the alert
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:108]:
```suggestion
```

File: modules/ROOT/pages/basic-alerts.adoc (hunk 2 above)
Comment (sarathecoubian) [reply modules/ROOT/pages/basic-alerts.adoc:118]: Removed and changed, respectively. Thank you!!

File: modules/ROOT/pages/basic-alerts.adoc [hunk 24]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+
+
+[%header%autowidth.spread]
+[[var-defs]]
+.Variable Definitions
+|===
+|Variable |Definition
+|`${alertLink}`|URL of the triggered alert
+|`${alertState}`|State of the alert when it was triggered. Available values are `OK` or `Alerting`.
+|`${comparison}`|Above or Below
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:103]:
```suggestion
|`${comparison}`|* Above, above or equal, equal, below or equal, below
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 25]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+|`${condition}`|Metric for which the alert is triggered
+|`${environment}`|Name of the environment in which the resource is deployed
```
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:105]:
```suggestion
|`${environment}`|Name of the environment in which the resource is deployed
|`${memberName}`|Name of the specific server that triggered the basic alert
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 26]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+== Basic Alert Email Template
+
+You can use the following variables in a basic alert email's *Subject* and *Message* fields.
+For the description of each variable, see <<var-defs>>.
+
+[%header,cols="25a,50a,20a"]
+.Variables for Alerts
+|===
+| Alert Type | Alert | Applicable Variables
+2+| All Alerts | `${resource}` `${severity}` `${state}`
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:53]:
```suggestion
2+| All Alerts | `${resource}` `${severity}`
```

Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:53]: All alerts support these variables

${severity}
${operator}
${threshold}
${period}
${value}
${condition}
${organization}
${environment}
${alertLink}
${dashboardLink}

Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:53]:
```suggestion
2+| All Alerts | ${severity}
${operator}
${threshold}
${period}
${value}
${condition}
${organization}
${environment}
${alertLink}
${dashboardLink}
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 27]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+.7+| CloudHub App Alerts |*CPU usage - CloudHub* .2+|`${operator}` `${value}`
+| *Memory usage - CloudHub*
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:55]: For cloudhub apps we support these metrics.

Cpu Utilization
Memory Utilization
Thread Count
Message Count
Message Error Count
Message Response Time

File: modules/ROOT/pages/basic-alerts.adoc [hunk 28]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+| *Custom application notification* | `${priority}` `${message}`
+| *Deployment failed* |  `${user}` `${message}`
+| *Deployment success* .2+| `${user}`
+| *Worker not responding*
+| *Exceeds event traffic threshold* | `${period}`
+
+
+.4+|Hybrid App Alerts | All Hybrid App Alerts| `${name}`
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:63]: For hybrid Apps we support alerts on these metrics
Message Count
Message Error Count
Message Response Time

File: modules/ROOT/pages/basic-alerts.adoc [hunk 29]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+| *Number of errors*
+| *Response time*
+
+.5+|Server Alerts | All Server Alerts | `${name}`
+| *Server CPU usage* .4+|  `${value}` `${comparison}` `${threshold}`
+| *Server Load average*
+| *Server Memory average*
+| *Server Thread count*
+
+.9+| Server Group Alerts| All Server Group Alerts | `${name}`
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:74]: For hybrid servers, server groups and clusters, these are the metrics we can create alerts on:

Cpu Utilization
Memory Utilization
Thread Count

File: modules/ROOT/pages/basic-alerts.adoc [hunk 30]
```diff
@@ -40,4 +40,79 @@ The following levels of severity are available for basic alerts:
...
+|`${environment}`|Name of the environment in which the resource is deployed
+|`${operator}` |An operator, such as `< > =`
+|`${organization}` |Name of the organization that owns the alert
+|`${period}` |Period of time over which to measure
+|`${resource}` a| Name of the item that triggers the alert:
+
+* CloudHub Apps: Name of the app that triggers the alert
+* Hybrid Apps: Name of the resource in the app that triggers the alert
+|`${severity}` | Severity of the alert
+|`${targetName}`|Name of the node in a server group or cluster
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:114]: targetName: Name of the server, server group, or cluster
memberName: Name of the server within the server group or cluster

**Comment by INameMyMethodsExactlyWhatTheyDo** [comment]
Lastly, I'm not sure how to show this on the docs, but the:
${resource} variable is only available for cloudhub, rtf, hybrid apps.
${targetName} variable is only available for hybrid apps, hybrid servers, server groups, clusters
${memberName} variable is only available for server groups and clusters.

File: modules/ROOT/pages/basic-alerts.adoc (hunk 30 above)
Comment (sarathecoubian) [inline modules/ROOT/pages/basic-alerts.adoc:114]:
```suggestion
|`${targetName}`|Name of the server, server group, or cluster
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 31]
```diff
@@ -40,4 +40,107 @@ The following levels of severity are available for basic alerts:
...
+* `${operator}`
+* `${threshold}`
+* `${period}`
+* `${value}`
+* `${condition}`
+* `${organization}`
+* `${environment}`
+* `${alertLink}`
+* `${dashboardLink}`
+* `${memberName}`
```
Comment (INameMyMethodsExactlyWhatTheyDo) [inline modules/ROOT/pages/basic-alerts.adoc:116]: Here ${memberName} does not apply to hybrid servers. 
We might need to have a separate server section apart from serverGroup and clusters

File: modules/ROOT/pages/basic-alerts.adoc [hunk 32]
```diff
@@ -40,4 +40,139 @@ The following levels of severity are available for basic alerts:
 
 include::partial$include-alert-severity.adoc[leveloffset=+1]
 
-// add basic alert email templates
+[[alert-variables]]
+== Basic Alert Email Template
+
+You can use the following variables in a basic alert email's *Subject* and *Message* fields.
```
Comment (hannanelson) [inline modules/ROOT/pages/basic-alerts.adoc:46]:
```suggestion
You can use the following variables in a basic alert email *Subject* and *Message* fields.
```

File: modules/ROOT/pages/basic-alerts.adoc [hunk 33]
```diff
@@ -40,4 +40,139 @@ The following levels of severity are available for basic alerts:
...
+|===
+
+
+[%header%autowidth.spread]
+[[var-defs]]
+.Variable Definitions
+|===
+|Variable |Definition
+|`${alertLink}`|URL of the triggered alert
+|`${alertState}`|State of the alert when it was triggered. Available values are `OK` or `Alerting`.
```
Comment (hannanelson) [inline modules/ROOT/pages/basic-alerts.adoc:161]:
```suggestion
|`${alertState}`|State of the alert when it was triggered.

Available values are `OK` or `Alerting`.
```

**Review by hannanelson — DISMISSED** [review]

**Review by hannanelson — APPROVED** [review]

**Review by hannanelson — APPROVED** [review]

**Pull request merged**